qutip/qip/pulse.py

```diff
@@ -179,12 +179,12 @@ class Pulse():
 
     Attributes
     ----------
-    ideal_pulse: :class:`qutip.qip.pulse._EvoElement`
+    ideal_pulse: :class:`._EvoElement`
```

Comment:
Something's very strange in the implementation if a public attribute is a private class like this.

Reply:
That's a fair point. I plan to replace it with `QobjEvo` at some point in a different PR. Make it private is a warning for that change... The difficulty is that `Pulse` does not have the `dim` info for the total Hamiltonian system. It just save the local Hamiltonian and the target qubits.

Reply:
It shows a problem with some part of the process - either here or in `QobjEvo`.  In a stricter language, you wouldn't even be able to access the type `_EvoElement` in this module, because it's not part of the public API.

I don't know what exactly to suggest (I'm fairly sure this PR isn't the one that introduced its usage), but using a class marked private is rather fragile.  The next person to edit `QobjEvo` may look at `_EvoElement` and assumed (somewhat correctly!) that they're completely free to change it, because it's not part of any API boundary.